packages/zowe-explorer/src/icons/IconGenerator.ts

```diff
@@ -0,0 +1,69 @@
+/**
+ * This program and the accompanying materials are made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Copyright Contributors to the Zowe Project.
+ *
+ */
+
+import { TreeItem } from "vscode";
+import { ZoweTreeNode } from "@zowe/zowe-explorer-api";
+import { Icon } from "../icons";
+
+export namespace IconGenerator {
+    export enum IconId {
+        "document" = "document",
+        "documentBinary" = "documentBinary",
+        "downloadedDocument" = "downloadedDocument",
+        "documentBinaryDownloaded" = "documentBinaryDownloaded",
+        "pattern" = "pattern",
+        "session" = "session",
+        "sessionInactive" = "sessionInactive",
+        "sessionActive" = "sessionActive",
+        "sessionActiveOpen" = "sessionActiveOpen",
+        "sessionOpen" = "sessionOpen",
+        "sessionFavourite" = "sessionFavourite",
+        "sessionFavouriteOpen" = "sessionFavouriteOpen",
+        "filterFolder" = "filterFolder",
+        "filterFolderOpen" = "filterFolderOpen",
+        "folder" = "folder",
+        "folderOpen" = "folderOpen",
+        "migrated" = "migrated",
+        "fileError" = "fileError",
+        "vsam" = "vsam",
+        "home" = "home",
+    }
+
+    export enum IconHierarchyType {
+        "base" = "base",
+        "derived" = "derived",
+    }
```

Comment:
Should we export these enums from top level of file to make them a bit easier to access? For example `IconId.document` instead of `IconGenerator.IconId.document`